infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -505,7 +495,7 @@ private CompactedEntry getCompactedEntries(String filePath,
                 LogEntry entry = LogEntry.parseFrom(logEntryBuf);
 
                 if (!noVerify) {
-                    if (metadata.getChecksum() != getChecksum(entry.toByteArray())) {
+                    if (metadata.getPayloadChecksum() != getChecksum(entry.toByteArray())) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1066)